cEP-0033.md

```diff
@@ -0,0 +1,760 @@
+the key is the name of the temporary file and the value is the contents of the
+linted file contents of the temporary file.
+
+We have a `assemble()` method inside the NlFileHandler, which uses
+the information from the `nl_sections` and extracts the sections from the
+`nl_diff_dict` and write it to the original file
+
+```python
+def assemble(nl_diff_dict, nl_sections, nl_info_dict):
+	"""
+	Assembles the temporary files. 
+
+	The sections are extracted by their increasing order of their index and 
+	then written directly to the original file.
+	"""
+	file = open(nl_info_dict['absolute_file_path'], 'w')
+
+	for nl_section in sorted(nl_sections.index):
+		temp_file_name = nl_section.temp_file
+		start = nl_section.linted_start
+		end = nl_section.linted_end
+		linted_file_content = nl_diff_dict[temp_file_name]
+
+		section_content = get_file_content(linted_file_content, start, end)
+
+		file.write(section_content)
+
+```
+
+## NlCore.py Prototype
+
+```python
+from coalib.nestedlib.NlInfoExtractor import extract_info
+from coalib.result_action.Diff import stats,get_diff_info
+from coalib.nestedlib.NlFileHandler import get_file_contents, get_nl_file_dict, assemble
+
+import coalib.nestedlib.Parser
+
+def get_arg_info(arg):
+	"""
+	Return argument list and nl_info_dict that  will be used to make 
+	the coala sections
+
+	>>> arg_list = get_arg_list(arg)
+	[ ( (file, temp_html_file), (bears, HTMLBear ), (settings, value) ),
+	  ( (file, temp_php_file), (bears, PHPBear ), (settings, value) )
+	]
+
+	"""
+    nl_info_dict = extract_info(arg)
+    arg_list = make_arg_list(nl_info_dict)
+
+	return nl_info_dict, arg_list
+
+def get_file_metadata(nl_info_dict):
+	"""
+	Returns the nl_sections and nl_file_dict
+	"""
+
+	parser = detect_parser(nl_info_dict['languages'])
+	file_contents = get_file_contents(nl_info_dict['absolute_file_path'])
+    nl_sections = parser.parse(file_contents)
+    nl_file_dict = get_nl_file_dict(nl_info_dict, nl_sections)
+
+	return nl_sections, nl_file_dict
+
+def find_section_index(diff_stats, diff_info, nl_sections):
+	"""
+	Returns the section index to which the patch is about to be applied.
+	"""
+
+	return index
+
+def update_nl_sections(diff_stats, diff_info, nl_sections):
+	"""
+	Updates the `linted_start` and the `linted_end` of the nl_sections 
+	"""
+
+	index = find_section_index(diff_stats, diff_info, nl_sections)
+
+	for nl_sections in nl_sections:
+		if nl_section.index == index:
+			"""
+			Update the values of linted_start` and the `linted_end`
+			"""
+
+
+def assemble_files(nl_diff_dict, nl_sections, nl_info_dict):
+	"""
+	Assembles the file and returns back to coala_main
+	"""
+
+	return assemble(nl_diff_dict, nl_sections, nl_info_dict)
+
+
+
+```
+
+## Changes in coala.py
+
+```python
+import coalib.nestedlib.Nlcore
+def main(debug=False):
+	configure_logging()
+	handle_nested = True
+
+    args = None  # to have args variable in except block when parse_args fails
```

Comment:
This sample's indentation is a bit messed up, so can't keep track of what's going where, but going by the comments, if you need args to be in the namespace, maybe include it as an argument for the main function